### PR TITLE
chore: migrate to tsdx

### PR DESCRIPTION
**Pull request by jaredhill4**
Perhaps the biggest changes here are in the configuration files in the project root.

The rest (including more than half) are moving the asset directory and cleaning up lint errors and warnings.

I just say that because, while 190 files seems overwhelming, the changes aren't too bad. :)

### CODE REVIEW

File: example/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "compilerOptions": {
+    "allowSyntheticDefaultImports": false,
+    "target": "es5",
```
Comment (dwene) [inline example/tsconfig.json:4]: is this the target we want?

File: src/lib/inputs/input.scss [hunk 2]
```diff
@@ -8,8 +8,8 @@
     input:not([type='color']) {
       box-sizing: content-box;
       height: 28px;
-      padding-top: 11px;
```
Comment (dwene) [inline src/lib/inputs/input.scss:11]: What was this change about? That will affect all our inputs right?

**Review by jaruesink — APPROVED** [review]

File: example/tsconfig.json (hunk 1 above)
Comment (jaredhill4) [reply example/tsconfig.json:4]: Note that this is in the `example` directory, so this isn't our `tsconfig.json`.

We actually don't include a `target` in our `tsconfig.json` - I just went with the `tsdx` default.

It looks like if you don't include a `target`, TS just defaults to `ES3`:

https://www.typescriptlang.org/tsconfig#target

Comment (jaredhill4) [reply example/tsconfig.json:4]: However, yes, the `target` we used previously was `es5`.

File: tsconfig.json [hunk 3]
```diff
@@ -1,20 +1,35 @@
 {
+  // see https://www.typescriptlang.org/tsconfig to better understand tsconfigs
+  "include": ["src", "types"],
   "compilerOptions": {
-    "rootDir": "src",
-    "target": "es5",
```
Comment (jaredhill4) [inline tsconfig.json:4]: @dwene, here is the previous target.

File: src/lib/inputs/input.scss (hunk 2 above)
Comment (jaredhill4) [reply src/lib/inputs/input.scss:11]: I adjusted this, because the padding was off in storybook.

File: example/tsconfig.json (hunk 1 above)
Comment (jaredhill4) [reply example/tsconfig.json]: I just went ahead and removed the `example` directory altogether.

**Pull request merged**